youtube_dl/extractor/nexx.py

```diff
@@ -313,8 +323,16 @@ def find_video(result):
             if result:
                 video = find_video(result)
 
+        if video:
+            # search for protection tokens -- if they're present we have to go
+            # through the procedure below...
+            tokenDASH = try_get(
+                video, lambda x: x['protectiondata']['tokenDASH'], compat_str)
+            tokenHLS = try_get(
+                video, lambda x: x['protectiondata']['tokenHLS'], compat_str)
```

Comment:
DRY.

Reply:
Both keys are used as indicators that the JSON metadata has to be re-fetched from a different location.  The values themselves are invalid.  They must be obtained from the *new* JSON metadata (s. above code).

Reply:
Again: you must eliminate code duplication. In both places.

Reply:
You have noticed my last two commits?